env0/data_provider.go

```diff
@@ -0,0 +1,58 @@
+package env0
+
+import (
+	"context"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataProvider() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataProviderRead,
+
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Type:         schema.TypeString,
+				Description:  "the type/name of the registry provider",
```

Comment:
no need for the word 'registry', as this API is only for the providers themselves